types/cstorclusterconfig.go

```diff
@@ -127,15 +127,24 @@ const (
 	PoolRAIDTypeDefault PoolRAIDType = PoolRAIDTypeMirror
 )
 
-// RAIDTypeToDefaultMinDiskCount maps pool instance's raid type
-// to its default minimum disk count
-var RAIDTypeToDefaultMinDiskCount = map[PoolRAIDType]int64{
+// RAIDTypeToDefaultDiskCount maps pool instance's raid type
+// to its default disk count
+var RAIDTypeToDefaultDiskCount = map[PoolRAIDType]int64{
 	PoolRAIDTypeMirror: 2,
 	PoolRAIDTypeStripe: 1,
 	PoolRAIDTypeRAIDZ:  3,
 	PoolRAIDTypeRAIDZ2: 6,
 }
 
+// RAIDTypeToMinimumDiskCount maps pool instance's raid type
+// to its minimum disk count
+var RAIDTypeToMinimumDiskCount = map[PoolRAIDType]int64{
+	PoolRAIDTypeMirror: 2,
+	PoolRAIDTypeStripe: 1,
+	PoolRAIDTypeRAIDZ:  3,
+	PoolRAIDTypeRAIDZ2: 4,
```

Comment:
is this 4 or 6?
cc @vishnuitta

Reply:
cc @avishnu

Reply:
There is no such minimum device count for a raid group. 
```
stripe = can have disk >=1 [but not more than 1 group] 
mirror = Should have exactly 2 disks in a group [raid group can be >=1]
raidz/raidz1 = can have 2^n+1  disks where n>0 in a group [raid group can be >=1]
raidz2 = can have 2^n+2 disks where n>=0 in a group [raid group can be >=1]
cc @vishnuitta

Reply:
Yes 4 is valid for raidz2.

Reply:
@avishnu -- A raidz2 raid group can have 6 disks also ...

Reply:
So are we trying to do something with a minimum ?

Reply:
Just putting info -- ( to avoid bugs ) 
If that is the case -- be aware that mirror raid group can have only 2 disks ( as a restriction imposed by control plane)
In this case -- if somehow disk count is 3 
The semantics of minimum here allows to pass the mirror validation but it should not-- not sure how implemented in the code, whether it is handled or not.

Reply:
Also what is the difference b/w -- `RAIDTypeToMinimumDiskCount` and `RAIDTypeToDefaultDiskCount`

Reply:
To be clear -- The minimum value for RAIDZ2 is 6 ( as a restriction imposed by control plane )
cc @AmitKumarDas @avishnu

Reply:
Above comment for raidz2  by me-- ( One mistake )
```
raidz2 = can have 2^n+2 disks where n>=0 in a group [raid group can be >=2]
```

Reply:
`RAIDTypeToMinimumDiskCount` -> minimum block device count in a raid group.
`RAIDTypeToDefaultDiskCount` -> If user not provided count in a group we are recommending this.

Reply:
Hmm... At the moment recommended and minimum will be same...
But good that we separated for we don't know when the recommended configuration might change.

Reply:
Good one shovan